index.js

```diff
@@ -60,10 +60,11 @@ function compression (options) {
     var length
     var listeners = []
     var stream
+    var noop = function () {}
 
-    var _end = res.end
+    res._end = res.end
```

Comment:
You're right that doing this isn't going to work, but I documented why I did it in the comment just above this one.